torch_dftd/torch_dftd3_calculator.py

```diff
@@ -92,19 +92,21 @@ def _calc_edge_index(
         )
 
     def _preprocess_atoms(self, atoms: Atoms) -> Dict[str, Optional[Tensor]]:
-        pos = torch.tensor(
-            atoms.get_positions(), device=self.device, dtype=self.dtype, requires_grad=True
-        )
+        pos = torch.tensor(atoms.get_positions(), device=self.device, dtype=self.dtype)
         Z = torch.tensor(atoms.get_atomic_numbers(), device=self.device)
         if any(atoms.pbc):
-            cell = torch.tensor(
-                atoms.get_cell(), device=self.device, dtype=self.dtype, requires_grad=True
+            cell: Optional[Tensor] = torch.tensor(
+                atoms.get_cell(), device=self.device, dtype=self.dtype
             )
         else:
             cell = None
         pbc = torch.tensor(atoms.pbc, device=self.device)
         edge_index, S = self._calc_edge_index(pos, cell, pbc)
-        input_dicts = dict(pos=pos, Z=Z, cell=cell, pbc=pbc, edge_index=edge_index, shift=S)
+        if cell is None:
+            shift = S
+        else:
+            shift = torch.mm(S, cell.detach())
```

Comment:
The definition of `shift` is different from the previous one. (`current_shift = previous_shift * cell`)
This is because gradient is calculated wrt. `shift` directly instead of cell.